srcs/app/http/HttpRequest.cpp

```diff
@@ -0,0 +1,195 @@
+#include <HttpRequest.hpp>
+
+const char HttpRequest::kCRLF_[] = "\r\n";
+const char HttpRequest::kWhitespace_[] = " \t";
+
+bool	HttpRequest::ParseRawString(const std::string &raw_request) {
+	offset_ = 0;
+	if (!ParseRequestLine_(raw_request))
+		return false;
+	if (!ParseHeaders_(raw_request))
+		return false;
+	if (!ParseBody_(raw_request))
+		return false;
+	return true;
+}
+
+std::string	HttpRequest::GetMethod() const {
+	return method_;
+}
+
+std::string	HttpRequest::GetUri() const {
+	return uri_;
+}
+
+std::string	HttpRequest::GetHttpVersion() const {
+	return http_version_;
+}
+
+std::string	HttpRequest::GetHeaderValue(const std::string &header_name) const {
+	const std::string			header_name_lc = ToLowerString(header_name);
+	HeadersMap::const_iterator	map_it = headers_.find(header_name_lc);
+	if (map_it != headers_.end())
+		return map_it->second;
```

Comment:
you can also use
```
try {
return headers_.at(header_name_lc)
} catch (...){}
return "";
```
instead of the whole iterator, find, if not foud etc.

Reply:
Good call, done.

Reply:
as far as I know, map.at() is c++98
```
alvaro@Tuor:~/.../practice00/cpp_module_02(master)# g++ --std=c++98 *.cpp
alvaro@Tuor:~/.../practice00/cpp_module_02(master)# clang++ --std=c++98 *.cpp
alvaro@Tuor:~/.../practice00/cpp_module_02(master)# grep "\.at" *
SpellBook.cpp:        delete _spells.at(spell);
SpellBook.cpp:        return _spells.at(spell);
TargetGenerator.cpp:        delete _targets.at(Target);
TargetGenerator.cpp:        return _targets.at(Target);
alvaro@Tuor:~/.../practice00/cpp_module_02(master)# grep -A 1 -B 6 "try" *
Binary file a.out matches
--
SpellBook.cpp-void SpellBook::learnSpell(ASpell *spell) {
SpellBook.cpp-    if (_spells.find(spell->getName()) == _spells.end())
SpellBook.cpp-        _spells[spell->getName()] = spell->clone();
SpellBook.cpp-}
SpellBook.cpp-
SpellBook.cpp-void SpellBook::forgetSpell(const std::string &spell) {
SpellBook.cpp:    try {
SpellBook.cpp-        delete _spells.at(spell);
SpellBook.cpp-        _spells.erase(spell);
SpellBook.cpp-    } catch (...){}
SpellBook.cpp-}
SpellBook.cpp-
SpellBook.cpp-ASpell *SpellBook::createSpell(std::string const &spell){
SpellBook.cpp:    try {
SpellBook.cpp-        return _spells.at(spell);
--
TargetGenerator.cpp-void TargetGenerator::learnTargetType(ATarget *Target) {
TargetGenerator.cpp-    if (_targets.find(Target->getType()) == _targets.end())
TargetGenerator.cpp-        _targets[Target->getType()] = Target->clone();
TargetGenerator.cpp-}
TargetGenerator.cpp-
TargetGenerator.cpp-void TargetGenerator::forgetTargetType(const std::string &Target) {
TargetGenerator.cpp:    try {
TargetGenerator.cpp-        delete _targets.at(Target);
TargetGenerator.cpp-        _targets.erase(Target);
TargetGenerator.cpp-    } catch (...){}
TargetGenerator.cpp-}
TargetGenerator.cpp-
TargetGenerator.cpp-ATarget *TargetGenerator::createTarget(std::string const &Target){
TargetGenerator.cpp:    try {
TargetGenerator.cpp-        return _targets.at(Target);
```

Reply:
You are right, the compilation error was related with another change that I made :+1: 
Resolved as discussed on Slack (its uncertain if `std::map.at()` is c++98 or c++11)